### ISSUE TITLE
GT miners should chunkload chunk they are mining- make sure algorithm is sane

### ISSUE THREAD

**Issue by richardhendricks** [GTNewHorizons/NewHorizons#2455]
GT miners should chunkload the chunk they are mining in.  No chunkloading of their own chunk - this way servers can manage their behavior properly.

Make sure chunkloading is server-friendly. 

Miner is supposed to mine downwards one layer at a time, but that may be prohibitive.

Maybe mine a 16x16x16 block and then step to the next one, and once a x16 layer is complete, drop the drill head 16 deeper?

**Comment by Dream-Master** [comment]
If you could add it in code? @richardhendricks

**Comment by Fapiko** [comment]
@richardhendricks What's the server overhead like with loading and unloading of a chunk? Is it significant enough that keeping the behavior as it is and just loading the chunk it's looking at is too sub-par, or even going 1 level per chunk at a time would have a performance impact?

**Comment by richardhendricks** [comment]
I don't know the overhead, but realize that as you tier up, the miners not only dig a wider area, but dig it faster.  And if they simply do an X/Z loop, they'll be jumping chunks every 16 blocks.  I don't know, I haven't seen the code yet to comment fully.

I don't want to do a full chunk and then move on, since the drill head is supposed to descend, but I am OK with it drilling out 16 layers and then descending to the next 16 layers.

That would be jumping chunks every 4096 blocks, which I think is way more reasonable.

**Comment by Dream-Master** [comment]
No the area got bigger each tier. @richardhendricks

**Comment by Fapiko** [comment]
@richardhendricks That makes sense, I am just wondering whether you're trying to optimize for a situation that isn't actually an issue or not. For example, if a server can handle running 100 of these rapidly loading/unloading chunks with no measurable performance impact, is it worthwhile to change the mining algorithm? I would say no.

I suspect that this probably isn't the case - but it might be something to look at if it simplifies things.

**Comment by richardhendricks** [comment]
The area gets bigger *and* speed is increased with each tier.  If these will only require 1 chunkloader where the miner is physically located, I expect more players would use them. I can't quantify server load, but it is better to try being nice now vs waiting for a server to brought to its knees.  TPS is already lower than 20 on both official servers AFAIK, so every tick will count.

Hmm...looks like the singleblock code *mines* a block each operation, or moves a layer down, and does it via a dumb X/Z loop. So it could potentially do many chunks in a single tick if the area is pretty empty. 

https://github.com/GTNewHorizons/GT5-Unofficial/blob/experimental/src/main/java/gregtech/common/tileentities/machines/basic/GT_MetaTileEntity_Miner.java#L114

**Comment by Dream-Master** [comment]
So possible to add or not @richardhendricks ?

**Comment by richardhendricks** [comment]
Low priority for now
